arweave/caip122.md

```diff
@@ -0,0 +1,71 @@
+---
+namespace-identifier: arweave-caip122
+title: Arweave Namespace - SIWx
+author: Rohit Pathare (@ropats16), Phil Billingsby (@pbillingsby), Dan MacDonald (@DanMacDonald)
+discussions-to: TBA
+status: Draft
+type: Standard
+created: 2022-09-01
+requires: ["CAIP-122", "CAIP-2", "CAIP-10"]
+---
+
+## CAIP-122
+
+For context, see the [CAIP-122](CAIP-122) specification.
+
+## Rationale
+
+On Arweave, [EIP-4361](EIP-4361) Sign-in with Arweave already exists, which was the inspiration for CAIP-122. This specification highlights how EIP-4361 conforms with CAIP-122.
+
+## Specification
+
+### Signing Algorithm
+
+Arweave wallets are RSA key pairs and use the private and public keys for signing and verifying signatures. Arweave wallet addresses are the SHA-256 hash of the RSA public key.
+
+### Signature Type
+
+Arweave signatures use the RSA-PSS signing algorithm to sign data and verify signatures.
+
+### Signature Creation
+
+The abstract data model must be converted into a string representation in an unambigious format. We use the format as defined in EIP-4361 as reference.
+
+An informal template of the full message is presented below. The field descriptions are provided in the order they must appear as follows:
+- `domain` is the RFC 3986 authority that is requesting the signing.
+- `address` is the Arweave address performing the signing.
+- `uri` is an RFC 3986 URI referring to the resource that is the subject of the signing (as in the subject of a claim).
+- `version` is the current version of the message, which MUST be 1 for this specification.
+- `type` of the signature to be generated, as defined in the namespaces for this CAIP.
+- `signature` is the result of concatenation and signing of the other field descriptions by the wallet.
```

Comment:
can it additionally include all field also in EIP-4361?

Reply:
As mentioned in the comment above, this CAIP is a first version of the standard for signing in with arweave and we are only using the EIP-4361 as a reference. Can having only the required fields for now suffice? We can add more fields as the standard becomes refined.

Reply:
ok, yeah in this case would include all required and optional fields, as most optional are needed to enable full use and options of CACAO, and we (ceramic) expect some optional fields to enable our use cases (ie resources, expiration), others may as well (ie nonce for authn)